include/function_info.h

```diff
@@ -46,15 +46,35 @@ namespace msl {
             using ret        = Ret;
             using cl         = Class;
             using args       = std::tuple<Args...>;
+            inline static constexpr size_t args_count = sizeof...(Args);
             using stack_args = std::tuple<
                                         std::remove_pointer_t<
                                             std::decay_t<Args>
                                         >...
                             >;
         };
 
+        template<typename ... TArgs>
+        void checkThanRefOrPointer()
+        {
+            static_assert(((std::is_reference_v<TArgs> || std::is_pointer_v<TArgs>) && ...), "must be pointer or reference");
+        }
+
     } /* end of namespace detal */
 
+// this macro will be unded in this file
```

Comment:
Maybe it's better to define this at the top, don't you think so?

Reply:
I think the macro should reap as little as possible

Reply:
Sorry for my wrong comment. If you still want i will move checkThanRefOrPointer at top. But it doesn't matter as it seems to me

Reply:
Well, it's definely the good idea to reduce macro "live-time". But just dislike having "macro magic" in namespaces, even if everything looks correct (as in our case). So, from my point of view we have to move it to the top.

Reply:
okay i will fix it.